src/futures/bufread/generic/decoder.rs

```diff
@@ -158,3 +158,25 @@ impl<R: AsyncBufRead, D: Decode> AsyncRead for Decoder<R, D> {
         }
     }
 }
+
+impl<R: AsyncBufRead + AsyncWrite, D: Decode> AsyncWrite for Decoder<R, D> {
```

Comment:
IMHO having a minimum generic bound would make writing generics using this impl easier:

```suggestion
impl<R: AsyncWrite, D> AsyncWrite for Decoder<R, D> {
```

Reply:
Thanks!